### ISSUE TITLE
Stuck Pending

### ISSUE THREAD

**Issue by AshleyAitken** [GoogleCloudPlatform/kubernetes#3349]
I have all the services running on each node (master and minions).

systemctl start docker.socket
systemctl start cadvisor.service
systemctl start etcd.service
systemctl start kube-apiserver.service
systemctl start kube-controller-manager.service
systemctl start kube-scheduler.service
systemctl start kube-proxy.service
systemctl start kubelet.service

I have run the Ansible setups (from remote machine).

hed0.private               : ok=26   changed=0    unreachable=0    failed=0   
hed1.private               : ok=15   changed=1    unreachable=0    failed=0   
hed2.private               : ok=15   changed=1    unreachable=0    failed=0 

But when I  do  /usr/bin/kubectl get pod fedoraapache

It never passes fedoraapache        fedora/apache       <unassigned>        name=fedoraapache   Pending

I can manually docker pull fedora/apache on the any of the nodes.

Thanks in advance for any assistance.

**Comment by dchen1107** [comment]
@scollier Do you have any idea about the issue? Thanks!

Ashley, could you please provide a little bit more information on your issue? which kubernete release? how to run 'kubectl get events | grep fedoraapache'?

**Comment by brendandburns** [comment]
Can you make sure that the machine name that the pod is scheduled to, is
the machine that the kubelet thinks it's running on.

We use `hostname` by default as the machine name from the kubelet's
perspective, so if you used ip address for your minions, there may be a
misscommunication between master and minion.

--brendan

**Comment by AshleyAitken** [comment]
Thank you very much for your posts @dchen1107 and @brendanburns 

I tried a lot of things but I think it may have been the hostname issue that was the problem. There were FQDN in some places and IP addresses elsewhere.  I change all to IP addresses and it works.

My general problems I believe, were caused by the fact that this does not seem to be made for  CentOS (7).

So all good now, and moved on to other issues (durable storage).
